subsys/net/lib/sockets/Kconfig

```diff
@@ -6,6 +6,7 @@
 menuconfig NET_SOCKETS
 	bool "BSD Sockets compatible API"
 	select FDTABLE
+	select REQUIRES_MALLOC if DNS_RESOLVER || NET_IP
```

Comment:
technically, getaddrinfo and freeaddrinfo could maybe use a slab if malloc is unavailable. What do you think @rlubos ? Not a change required here, but maybe a good future enhancement. Even malloc-less systems should be able to resolve.